packages/ses/src/compartment.js

```diff
@@ -359,24 +360,29 @@ export const makeCompartmentConstructor = (
      * The method `compartment.import` accepts a full specifier, but dynamic
      * import accepts an import specifier and resolves it to a full specifier
      * relative to the calling module's full specifier.
+     * @param compartment - The `compartment` reference, to avoid relying on upper-scope `this`
      * @returns {Promise<ModuleExportsNamespace>}
      */
-    const compartmentImport = async fullSpecifier => {
-      if (typeof resolveHook !== 'function') {
-        throw TypeError(
-          `Compartment does not support dynamic import: no configured resolveHook for compartment ${q(name)}`,
+    const compartmentImport = functionBind(
```

Comment:
We can obviate `functionBind` here by capturing `const compartment = this` high up the `Compartment` constructor, and removing the positional `compartment` argument.